### PR TITLE
Explore layout and course search

### PR DESCRIPTION
**Pull request by ChrisXJH**
## Demo
search

### CODE REVIEW

File: src/components/ExploreLayout/index.js [hunk 1]
```diff
@@ -1,102 +1,31 @@
...
-        loading={loading}
-      />
-    </Paper>
+    <Container maxWidth="xl" className={classes.layoutContainer}>
+      <Paper className={classes.searchBoxContainer}>
+        <SubjectsBar />
+        <Divider />
+        <SearchField
+          className={classes.searchField}
+          onChange={_.debounce(handleSearchChange, 300)}
```
Comment (colelin26) [inline src/components/ExploreLayout/index.js:22]: I am curious. Is the delay here used to solve the race conditions?

File: src/services/CourseService.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-underscore-dangle */
+import LRU from 'lru-cache';
+import Fuse from 'fuse.js';
+import Course from '../data/models/Course';
+
+const fuseOptions = {
```
Comment (colelin26) [inline src/services/CourseService.js:6]: I spent some time playing the options here with `distance, location, threshold` and yah it is quite difficult to get the score in fuse right. Maybe we can add a regex matching for `subject catalogNum` in the future to make `CS350` the first when the user is just searching for `CS350`.

File: src/components/ExploreLayout/index.js (hunk 1 above)
Comment (ChrisXJH) [reply src/components/ExploreLayout/index.js:22]: This is not for race condition. It's used to reduce the number of search requests when the user is typing.

File: src/services/CourseService.js (hunk 2 above)
Comment (ChrisXJH) [reply src/services/CourseService.js:6]: I've looked into the `tag` option. We can generate a `tag` field for each course. So far it's working ok with this simple `fuseOptions`. We can look into that later.

**Pull request merged**